packages/statemanager/src/stateManager.ts

```diff
@@ -926,6 +926,9 @@ export class DefaultStateManager implements EVMStateManagerInterface {
    * 2. Cache values are generally not copied along
    */
   shallowCopy(): DefaultStateManager {
+    const common = this.common.copy()
+    common.setHardfork(this.common.hardfork())
```

Comment:
We need to `setHardfork` here to initialize the cache, right?

Reply:
Yes, I think it's at least a bit safer like this (not 1000% sure it's necessary).